src/views/toolbar/data-toolbar-directive.js

```diff
@@ -0,0 +1,248 @@
+
+      var matchesFilter = function (item, filter) {
+        var match = true;
+
+        if (filter.id === 'name') {
+          match = item.name.match(filter.value) !== null;
+        } else if (filter.id === 'address') {
+          match = item.address.match(filter.value) !== null;
+        } else if (filter.id === 'birthMonth') {
+          match = item.birthMonth === filter.value;
+        }
+        return match;
+      };
+
+      var matchesFilters = function (item, filters) {
+        var matches = true;
+
+        filters.forEach(function(filter) {
+          if (!matchesFilter(item, filter)) {
+            matches = false;
+            return false;
+          }
+        });
+        return matches;
+      };
+
+      var applyFilters = function (filters) {
+        $scope.items = [];
+        if (filters && filters.length > 0) {
+          $scope.allItems.forEach(function (item) {
+            if (matchesFilters(item, filters)) {
+              $scope.items.push(item);
+            }
+          });
+        } else {
+          $scope.items = $scope.allItems;
+        }
+      };
+
+      var filterChange = function (filters) {
+      $scope.filtersText = "";
+        filters.forEach(function (filter) {
+          $scope.filtersText += filter.title + " : " + filter.value + "\n";
+        });
+        applyFilters(filters);
+        $scope.toolbarConfig.filterConfig.resultsCount = $scope.items.length;
+      };
+
+      $scope.filterConfig = {
+        fields: [
+          {
+            id: 'name',
+            title:  'Name',
+            placeholder: 'Filter by Name',
+            filterType: 'text'
+          },
+          {
+            id: 'address',
+            title:  'Address',
+            placeholder: 'Filter by Address',
+            filterType: 'text'
+          },
+          {
+            id: 'birthMonth',
+            title:  'Birth Month',
+            placeholder: 'Filter by Birth Month',
+            filterType: 'select',
+            filterValues: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
+          }
+        ],
+        resultsCount: $scope.items.length,
+        appliedFilters: [],
+        onFilterChange: filterChange
+      };
+
+      var viewSelected = function(viewId) {
+        $scope.viewType = viewId
+      };
+
+      $scope.viewsConfig = {
+        views: [pfViewUtils.getListView(), pfViewUtils.getTilesView()],
+        onViewSelect: viewSelected
+      };
+      $scope.viewsConfig.currentView = $scope.viewsConfig.views[0].id;
+      $scope.viewType = $scope.viewsConfig.currentView;
+
+      $scope.toolbarConfig = {
+        viewsConfig: $scope.viewsConfig,
+        filterConfig: $scope.filterConfig
+      };
+
+      $scope.listConfig = {
+        selectionMatchProp: 'name',
+        checkDisabled: false
+      };
+    }
+  ]);
+  </file>
+</example>
+ */
+angular.module('patternfly.views').directive('pfDataToolbar',
```

Comment:
I don't think a 'toolbar' should be under 'views', I know it consumes the 'views' mechanism, but it should probably go into a general 'components' directive.  For now it's fine.